lib/spack/llnl/util/filesystem.py

```diff
@@ -1415,6 +1415,6 @@ def search_paths_for_executables(*path_hints):
 
         bin_dir = os.path.join(path, 'bin')
         if os.path.isdir(bin_dir):
-            executable_paths.append(bin_dir)
+            executable_paths.append(os.path.abspath(bin_dir))
```

Comment:
This works if the user gives a `PATH` like `../gcc/` but not if the user gives a `PATH` like `../gcc/bin`. I think this logic should be moved farther up.

Reply:
Oh good call. I have pushed a change that fixes it.